rfc/egress-with-ucan.md

```diff
@@ -0,0 +1,159 @@
+
+### Format
+
+The format for the `did:bearer` method conforms to the [DID specification](https://www.w3.org/TR/did-core/). It consists of the `did:bearer` prefix, followed by the percent-encoded token. The ABNF for the format is described below, using the ABNF syntax described in [RFC 5234](https://www.rfc-editor.org/rfc/rfc5234). The definition of `idchar` is found in the [DID syntax](https://www.w3.org/TR/did-core/#did-syntax).
+
+```rfc5234
+did-bearer-format = "did:bearer:" idchar
+```
+
+### Operations
+
+The following section outlines the DID operations for the `did:bearer` method.
+
+#### Create
+
+Creating a `did:bearer` value begins with a token (a string) which it should represent.
+
+1. Percent-encode the token value, as specified in [RFC 3986 Section 2.1](https://www.rfc-editor.org/rfc/rfc3986#section-2.1).
+2. Prepend the result with `did:bearer:`.
+
+For example, the token `abc$*)123` would be represented by the DID `did:bearer:abc%24%2a%29123`. Anyone presenting the token `abc$*)123` by any means a service accepts may be considered by that service to be the [DID subject](https://www.w3.org/TR/did-core/#dfn-did-subjects) of `did:bearer:abc%24%2a%29123`.
+
+The DID Document for a `did:bearer` DID is extremely simple. A `did:bearer` DID is associated with no cryptographic keys. It is not capable of signing. All authentication occurs out of band, so it has no verification methods. The DID Document is therefore trivially generated from the DID. An example is given below that expands the DID `did:bearer:abcde12345` into its associated DID Document:
+
+```json
+{
+  "@context": "https://www.w3.org/ns/did/v1",
+  "id": "did:bearer:abcde12345"
+}
+```
+
+#### Read
+
+Reading a `did:bearer` value is a matter of deterministically expanding the value to a DID Document. This process is described above in [Create](#create).
+
+#### Update
+
+This DID Method does not support updating the DID Document.
+
+#### Deactivate
+
+This DID Method does not support deactivating the DID Document.
+
+## Making content retrievable with a token
+
+A UCAN delegation is used to allow token access to content. Typically, this is performed by an Agent within a Storacha Client, who has the authority to `/space/content/retrieve` from the Space by way of logging into an Account—thus, the delegation chain flows Space → Account → Agent → Token.
+
+```json5
+// UCAN 1.0
+{
+  "iss": "did:key:zAgent",
+  "aud": "did:bearer:abcde12345",
+  "sub": "did:key:zSpace",
+  "cmd": "/space/content/retrieve",
+  "pol": [
+    ["==", ".cid", "bafy...7pcu"],
+  ],
+  "nonce": …,
+  "exp": …
+}
+```
+
+```json5
+// UCAN 0.9
+{
+  "iss": "did:key:zAgent",
+  "aud": "did:bearer:abcde12345",
+  "att": [
+    {
+      "with": "did:key:zSpace",
+      "can": "space/content/retrieve",
+      "nb": {
+        "cid": "bafy...7pcu"
```

Comment:
I don't think we do want that to be a link, actually. We want to request the content using the CID as an argument; this would provide the content itself (by reference) as the argument. Importantly, I believe, policies would resolve against the resolved `bafy...7pcu`, not the string `"bafy...7pcu"`. We want to match on the string.

Reply:
There's no obligation to include the content in the invocation - we don't for upload/add for example.

I'm almost certain you can match a link in a policy...

IDK maybe I'm not understanding right?

Reply:
The spec doesn't appear to say explicitly, but [it does say](https://github.com/ucan-wg/delegation/blob/0dbaca11cdd0c6a5ccb89d77db3f5fb0d2245347/README.md#selecting-on-bytes):

> ### Selecting on Bytes
>
> Bytes MAY be selected into. When doing so, they MUST be treated as a byte array (`[u8]`), and MUST NOT be treated as a Base64 string or any other representation.
>
> ```source-js
> // DAG-JSON
> { "/": { "bytes": "1qnBjPjE" } }
> 
> // Hexadecimal
> 0xd6 0xa9 0xc1 0x8c 0xf8 0xc4
> 
> // Selector
> ".[3]"
> // ⬆️  0x8c = 140
> ```

If the policy resolver understands DAG-JSON bytes, I assume it understands links as well. I'm not sure what the correct behavior would be when matching on a link, but I don't think it would be to treat it as a string, or as a literal JSON map `{ "/": "bafy...foo" }`. My assumption would be that, if anything, it would attempt to inline the link before applying the policy, and perhaps bail if it didn't have access to the content.